business/workloads.go

```diff
@@ -613,6 +635,8 @@ func fetchWorkloads(ctx context.Context, layer *Layer, namespace string, labelSe
 		return nil, err
 	}
 
+	kialiClient := clientFactory.GetSAClient(cluster)
```

Comment:
I think we need to use the user client here?, to just get what the user is allowed to see?.

Reply:
@leandroberetta thanks, right, will use the user client once https://github.com/kiali/kiali/pull/5851 get merged, for now the main point of this PR is to rework workloads list and details to add "cluster" property.